code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -0,0 +1,390 @@
+	if(!user.melee_damage_upper && !user.obj_damage)
+		user.emote("custom", message = "[user.friendly_verb_continuous] [src].")
+		return 0
+	else
+		var/play_soundeffect = 1
+		if(user.environment_smash)
+			play_soundeffect = 0
+			playsound(src, 'sound/effects/bang.ogg', 50, TRUE)
+		var/animal_damage = rand(user.melee_damage_lower,user.melee_damage_upper)
+		if(user.obj_damage)
+			animal_damage = user.obj_damage
+		animal_damage = min(animal_damage, 20*user.environment_smash)
+		log_combat(user, src, "attacked")
+		attack_generic(user, animal_damage, user.melee_damage_type, MELEE, play_soundeffect)
+		return 1
+
+
+/obj/vehicle/sealed/mecha/hulk_damage()
+	return 15
+
+/obj/vehicle/sealed/mecha/attack_hulk(mob/living/carbon/human/user)
+	. = ..()
+	if(.)
+		log_message("Attack by hulk. Attacker - [user].", LOG_MECHA, color="red")
+		log_combat(user, src, "punched", "hulk powers")
+
+/obj/vehicle/sealed/mecha/blob_act(obj/structure/blob/B)
+	log_message("Attack by blob. Attacker - [B].", LOG_MECHA, color="red")
+	take_damage(30, BRUTE, MELEE, 0, get_dir(src, B))
+
+/obj/vehicle/sealed/mecha/attack_tk()
+	return
+
+/obj/vehicle/sealed/mecha/hitby(atom/movable/AM, skipcatch, hitpush, blocked, datum/thrownthing/throwingdatum) //wrapper
+	log_message("Hit by [AM].", LOG_MECHA, color="red")
+	. = ..()
+
+/obj/vehicle/sealed/mecha/bullet_act(obj/item/projectile/Proj) //wrapper
+	if(!enclosed && LAZYLEN(occupants) && !(mecha_flags  & SILICON_PILOT) && !Proj.force_hit && (Proj.def_zone == BODY_ZONE_HEAD || Proj.def_zone == BODY_ZONE_CHEST)) //allows bullets to hit the pilot of open-canopy mechs
+		for(var/m in occupants)
+			var/mob/living/hitmob = m
+			hitmob.bullet_act(Proj) //If the sides are open, the occupant can be hit
+		return BULLET_ACT_HIT
+	log_message("Hit by projectile. Type: [Proj.name]([Proj.flag]).", LOG_MECHA, color="red")
+	. = ..()
+
+/obj/vehicle/sealed/mecha/ex_act(severity, target)
+	log_message("Affected by explosion of severity: [severity].", LOG_MECHA, color="red")
+	if(prob(deflect_chance))
+		severity++
+		log_message("Armor saved, changing severity to [severity]", LOG_MECHA)
+	. = ..()
+
+
+/obj/vehicle/sealed/mecha/contents_explosion(severity, target, origin)
+	severity++
+	for(var/X in equipment)
+		var/obj/item/mecha_parts/mecha_equipment/ME = X
+		ME.ex_act(severity, target, origin)
+	for(var/Y in trackers)
+		var/obj/item/mecha_parts/mecha_tracking/MT = Y
+		MT.ex_act(severity, target, origin)
+	for(var/Z in occupants)
+		var/mob/living/occupant = Z
+		occupant.ex_act(severity, target, origin)
+
+/obj/vehicle/sealed/mecha/handle_atom_del(atom/A)
+	if(A in occupants)
+		LAZYREMOVE(occupants, A)
+		icon_state = initial(icon_state)+"-open"
+		setDir(dir_in)
+
+/obj/vehicle/sealed/mecha/emp_act(severity)
+	. = ..()
+	if (. & EMP_PROTECT_SELF)
+		return
+	if(get_charge())
+		use_power((cell.charge/3)/(severity*2))
+		take_damage(30 / severity, BURN, ENERGY, 1)
+	log_message("EMP detected", LOG_MECHA, color="red")
```

Comment:
Our EMPs use severities from 0 (weak) - 100 (strong) as opposed to 1 (heavy) - 2 (light)

Reply:
I'll attempt to restore original behavior.